src/iterator.rs

```diff
@@ -274,6 +286,13 @@ impl<'a> Iterator<'a> {
 
     // Advances the iterator by one.
     pub fn next(&mut self) {
+        let key = BytesMut::from(&self.item().key.clone()[..]);
+        let key_with_ts = key_with_ts(key, self.item().version);
+
+        // TODO: Consider add direction flag to avoid seek.
+        self.table_iter.seek(&key_with_ts);
```

Comment:
why seek

Reply:
Since `parse_item` and `parse_item_for_prev` would advance and retreat the `table_iter` inside the `Iterator` respectively, what we should do in case of of `next` after `next` and `prev` after `next` are different. We may add flag or update `parse_item` and `parse_item_for_prev` to avoid seek.

Reply:
I added a flag `current_direction` to indicate the "direction" of the iterator. If the last operation is `prev`, it will be set to false, and we will advance the inner iterator double times in that case if we call `next` subsequently.